### PR TITLE
Reworked frontend for order component and order bike

### LINKED ISSUE #166: Rework order bike and component page on frontend

### PR DESCRIPTION
**Pull request by nasaku898**
closes #166 

image

image
image

### CODE REVIEW

**Review by huangs08 — APPROVED** [review]

File: bike-erp/backend/src/main/dao/ComponentDAO.ts [hunk 1]
```diff
@@ -46,6 +46,15 @@ export const fetchComponentLocation = (id: string) => {
...
 
+export const fetchAllLocations = () => {
+  return new Promise((resolve, reject) => {
+    const query = "select distinct location_name from component_location"
+    db.query(query, (err, rows) => {
+      if (err) return reject(err);
+      resolve(JSON.parse(JSON.stringify(rows)))
+    })
+  })
+}
```
Comment (Manraj45) [inline bike-erp/backend/src/main/dao/ComponentDAO.ts:57]: skip line after this

File: bike-erp/src/components/BikeOrderSummary/BikeOrderSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import { RemoveCircleOutline } from "@material-ui/icons"
+import { useState } from "react"
+import { connect } from "react-redux"
+import { BACKEND_URL } from "../../core/utils/config"
+import { BikeSold, removeAllBikes, removeAllComponents, removeBike, removeComponentSold } from "../../redux/actions/OrderBikeActions/orderBikeActions"
+import useStyles from "./BikeOrderSummaryStyle"
+import RemoveIcon from '@material-ui/icons/Remove';
+import AddIcon from '@material-ui/icons/Add';
+import axios from "axios"
+import { useEffect } from "react"
```
Comment (Manraj45) [inline bike-erp/src/components/BikeOrderSummary/BikeOrderSummary.tsx:11]: Skip line and add header comment for file

File: bike-erp/src/components/BikeOrderSummary/BikeOrderSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    useEffect(() => {
+        let cartTotal = 0
+        for (let i = 0; i < bikeOrderList.bikeOrderList.length; i++) {
+            cartTotal = cartTotal + bikeOrderList.bikeOrderList[i].price * bikeOrderList.bikeOrderList[i].quantity
+            if (bikeOrderList.bikeOrderList.length - 1 === i) {
+                setTotal(cartTotal)
+            }
+        }
+    }, [bikeOrderList.bikeOrderList])
```
Comment (Manraj45) [inline bike-erp/src/components/BikeOrderSummary/BikeOrderSummary.tsx:25]: skip line here and between all the const

File: bike-erp/src/components/OrderSummary/OrderSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import { connect } from "react-redux"
+import { removeAllItem, changeQuantity, deleteItemFromCart } from "../../redux/actions/OrderListActions/orderListAction"
+import useStyles from "./OrderSummaryStyle"
+import AddIcon from '@material-ui/icons/Add';
+import RemoveIcon from '@material-ui/icons/Remove';
+import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';
+import axios from "axios"
+import { BACKEND_URL } from "../../core/utils/config"
+
+const OrderSummary = ({ orderList, removeItem, removeAllItems, updateQuantity }: any) => {
```
Comment (Manraj45) [inline bike-erp/src/components/OrderSummary/OrderSummary.tsx:12]: Add header comment for the file

File: bike-erp/src/pages/OrderBike/OrderBikeStyle.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { makeStyles } from "@material-ui/core";
+
+const useStyles = makeStyles((theme) => ({
+
```
Comment (Manraj45) [inline bike-erp/src/pages/OrderBike/OrderBikeStyle.ts:4]: remove empty line here

File: bike-erp/src/pages/OrderComponent/OrderComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    }, [selectSize, location, URL, orderList.orderList])
+
+    const checkForDuplicateItem = (order: Order, selectedId: number) => {
+        return (order.id !== selectedId);
+    }
+
+    const handleAddToCart = () => {
+        axios.get(`${URL}/components/${selected}`).then(response => {
+            const retrievedComponent = response.data[0]
+            // if (retrievedComponent.quantity <= 0) {
```
Comment (Manraj45) [inline bike-erp/src/pages/OrderComponent/OrderComponent.tsx:80]: Remove commented line

File: bike-erp/src/pages/OrderComponent/OrderComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+import axios from "axios"
+import { useEffect, useState } from "react"
+import { BACKEND_URL } from "../../core/utils/config"
+import { withStyles } from '@material-ui/core/styles';
+import useStyles from "./OrderComponentStyle"
+import { connect } from "react-redux"
+import { addItem, Order } from "../../redux/actions/OrderListActions/orderListAction"
+import OrderSummary from "../../components/OrderSummary/OrderSummary"
+
+const OrderComponent = ({ addItem, orderList }) => {
```
Comment (Manraj45) [inline bike-erp/src/pages/OrderComponent/OrderComponent.tsx:12]: Add header comment for file

File: bike-erp/src/pages/OrderComponent/OrderComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+    const classes = useStyles()
+
+    useEffect(() => {
+        axios.get(`${URL}/components/locations/all`,
+        ).then((response) => {
+            setLocationList(response.data)
+        }).catch(error => {
+            //handle error
+            console.log(error)
```
Comment (Manraj45) [inline bike-erp/src/pages/OrderComponent/OrderComponent.tsx:48]: remove console.log here and maybe handle error?

File: bike-erp/src/pages/OrderComponent/OrderComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+                    {
+                        params: {
+                            location: location,
+                            size: selectSize
+                        }
+                    }).then(response => {
+                        setComponentList(response.data)
+                    }).catch(error => {
+                        //handle error after
+                        console.log(error)
```
Comment (Manraj45) [inline bike-erp/src/pages/OrderComponent/OrderComponent.tsx:65]: same here

File: bike-erp/src/pages/OrderComponent/OrderComponentStyle.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import { makeStyles } from "@material-ui/core";
+
+const useStyles = makeStyles((theme) => ({
+
```
Comment (Manraj45) [inline bike-erp/src/pages/OrderComponent/OrderComponentStyle.ts:4]: remove extra space here

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Good job! Just some minor changes

**Review by Manraj45 — APPROVED** [review]
Good job!

**Pull request merged**